### PR TITLE
Test, benchmarks & fixes to UTF-8 validation

### PR DESCRIPTION
**Pull request by chfast**
Includes 32-bit architecture and dead store fixes.
Also added "UTF-8 encoded sample plain-text file" by Markus Kuhn. Used to testing and benchmarking.

```
parse_string/16_mean                    +0.0013         +0.0013            31            31            31            31
parse_string/32_mean                    +0.0007         +0.0007            30            30            30            30
parse_string/64_mean                    +0.0016         +0.0016            47            47            47            47
parse_string/128_mean                   +0.0003         +0.0003            68            68            68            68
parse_string/256_mean                   +0.0000         +0.0000           109           109           109           109
parse_string/512_mean                   -0.0003         -0.0003           192           192           192           192
parse_string/1024_mean                  +0.0001         +0.0001           362           362           362           362
parse_string/2048_mean                  -0.0009         -0.0009           720           719           720           719
parse_string/4096_mean                  -0.0992         -0.0992          1576          1420          1576          1420
utf8_demo_mean                          +0.0099         +0.0099         16991         17159         16991         17159
```

### CODE REVIEW

File: test/utils/utf8_demo.cpp [hunk 1]
```diff
@@ -0,0 +1,222 @@
+// The content of the "UTF-8 encoded sample plain-text file" by Markus Kuhn.
+// SPDX-License-Identifier: CC-BY-4.0
+#include "utf8_demo.hpp"
+
+namespace fizzy::test
+{
+const std::string_view utf8_demo =
```
Comment (axic) [inline test/utils/utf8_demo.cpp:7]: Isn't string_view a reference?

File: test/utils/utf8_demo.cpp [hunk 2]
```diff
@@ -0,0 +1,222 @@
+// The content of the "UTF-8 encoded sample plain-text file" by Markus Kuhn.
```
Comment (axic) [inline test/utils/utf8_demo.cpp:1]: Do you have a link?

File: circle.yml [hunk 3]
```diff
@@ -320,7 +320,7 @@ jobs:
       - run:
           name: "Run codespell"
           command: |
-            codespell --quiet-level=4 -I .codespell-whitelist
+            codespell --quiet-level=4 --ignore-words .codespell-whitelist --skip 'utf8_demo.cpp'
```
Comment (axic) [inline circle.yml:323]: What does `--ignore-words` do?

Comment (chfast) [reply circle.yml:323]: It's expanded `-I` option.

File: test/utils/utf8_demo.cpp (hunk 2 above)
Comment (chfast) [reply test/utils/utf8_demo.cpp:1]: I could not find the original file, but the content has license id and link to author's home page. Many projects uses this file, including:
- https://github.com/cyb70289/utf8
- https://www.w3.org/2001/06/utf-8-test/UTF-8-demo.html (older version).

File: test/utils/utf8_demo.cpp (hunk 1 above)
Comment (chfast) [reply test/utils/utf8_demo.cpp:7]: It is, but reference to a literal is fine.

File: test/utils/utf8_demo.cpp (hunk 2 above)
Comment (chfast) [reply test/utils/utf8_demo.cpp:1]: Actually, I have found it at https://www.cl.cam.ac.uk/~mgk25/ucs/examples/UTF-8-demo.txt.

**Review by axic — APPROVED** [review]

**Review by gumb0 — APPROVED** [review]

**Pull request merged**